### PR TITLE
fix harmless warnings

### PR DESCRIPTION
**Pull request by toxieainc**
this makes the code compile out of the box with MSVC, plus fixes some warnings found with PVS

### CODE REVIEW

File: src/ymfm.h [hunk 1]
```diff
@@ -136,7 +140,7 @@ inline uint8_t count_leading_zeros(uint32_t value)
 inline uint8_t count_leading_zeros(uint32_t value)
 {
 	unsigned long index;
-	return _BitScanReverse(&index, value) ? (31U - index) : 32U;
+	return _BitScanReverse(&index, value) ? (uint8_t)(31U - index) : 32U;
 }
```
Comment (aaronsgiles) [inline src/ymfm.h:144]: Please use C++-style casts, e.g., uint8_t(31U - index), for arithmetic casts.

File: src/ymfm.h [hunk 2]
```diff
@@ -355,8 +359,8 @@ class ymfm_saved_state
 	void save(bool &data) { write(data ? 1 : 0); }
 	void save(int8_t &data) { write(data); }
 	void save(uint8_t &data) { write(data); }
-	void save(int16_t &data) { write(data).write(data >> 8); }
-	void save(uint16_t &data) { write(data).write(data >> 8); }
+	void save(int16_t &data) { write((uint8_t)data).write(data >> 8); }
+	void save(uint16_t &data) { write((uint8_t)data).write(data >> 8); }
 	void save(int32_t &data) { write(data).write(data >> 8).write(data >> 16).write(data >> 24); }
```
Comment (aaronsgiles) [inline src/ymfm.h:364]: Please use C++-style casts for this.

File: src/ymfm_pcm.cpp [hunk 3]
```diff
@@ -425,7 +425,7 @@ void pcm_channel::load_wavetable()
 	// for some reason that is unclear
 	m_endpos = read_pcm(wavheader + 5) << 8;
 	m_endpos |= read_pcm(wavheader + 6);
-	m_endpos = -m_endpos << 16;
+	m_endpos = -(int32_t)m_endpos << 16;
```
Comment (aaronsgiles) [inline src/ymfm_pcm.cpp:429]: One more spot to use C++-style cast.

**Review by aaronsgiles — CHANGES_REQUESTED** [review]

**Comment by toxieainc** [comment]
Done.

**Pull request merged**